cmake/kokkos_arch.cmake

```diff
@@ -622,10 +622,15 @@ IF (KOKKOS_ENABLE_SYCL)
   COMPILER_SPECIFIC_OPTIONS(
     DEFAULT -fsycl-unnamed-lambda
   )
-  IF (NOT KOKKOS_ENABLE_IMPL_SYCL_RELOCATABLE_DEVICE_CODE)
-    IF(KOKKOS_CXX_COMPILER_ID STREQUAL IntelLLVM AND KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 2023.1.0)
-      MESSAGE(FATAL_ERROR "Kokkos_ENABLE_IMPL_SYCL_RELOCATABLE_DEVICE_CODE=OFF requires oneAPI 2023.1.0 or later")
+  IF(KOKKOS_CXX_COMPILER_ID STREQUAL IntelLLVM AND KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 2023.1.0)
+    # Before oneAPI 2023.1.0 the compiler didn't support -f[no-]sycl-rdc
+    # but allowed separate compilation by default.
+    IF(NOT KOKKOS_ENABLE_SYCL_RELOCATABLE_DEVICE_CODE)
+      MESSAGE(FATAL_ERROR "Kokkos_ENABLE_SYCL_RELOCATABLE_DEVICE_CODE=OFF requires oneAPI 2023.1.0 or later")
```

Comment:
Does SYCL_EXTERN exist in older SYCL versions and does it work?

Reply:
Yes, the CI is using oneAPI 2023.0.0 and builds and runs the example fine.